src/scib_metrics/utils/_pcr.py

```diff
@@ -38,6 +40,10 @@ def principal_component_regression(
         raise ValueError("Dimension mismatch: X must be 2-dimensional.")
     if X.shape[0] != covariate.shape[0]:
         raise ValueError("Dimension mismatch: X and batch must have the same number of samples.")
+    if categorical:
+        covariate = np.asarray(pd.Categorical(covariate).codes)
+    else:
+        covariate = np.asarray(covariate)
 
     covariate = one_hot(covariate) if categorical else covariate.reshape((covariate.shape[0], 1))
```

Comment:
```suggestion
```

Reply:
oh didn't see this. I will make the change!